src/NuGet.Server.V2/Controllers/NuGetODataController.cs

```diff
@@ -0,0 +1,315 @@
+﻿using NuGet.Server.Core.DataServices;
+using NuGet.Server.Core.Infrastructure;
+using NuGet.Server.V2.OData;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Net.Mime;
+using System.Threading.Tasks;
+using System.Web.Http;
+using System.Web.Http.OData;
+using System.Web.Http.OData.Query;
+
+namespace NuGet.Server.V2.Controllers
+{
+    [NuGetODataControllerConfiguration]
+    public abstract class NuGetODataController : ODataController
```

Comment:
The other conventions are needed for returning IActionResult everywhere, which is what we'd like to have for NuGet.Server as well. Reason is that when returning IActionResult, unit tests actually run the entire OData pipeline (as done in the gallery). In this current PR, testing would be checking IQueryables, but those are the queryables as seen before OData does its magic.